### DISCUSSION TITLE
How do I call the Magento API SOAP V2 with multiple complex filters to the same field?

### DISCUSSION THREAD

**Discussion by growit-chris** [OpenMage/magento-lts#2116] [Questions & Answers]
Since the Magento community in StackExchange seems to be dead (at least for version 1.x), I ask my question here. I hope you can help me. After all, it's a 'problem' of the Magento API. ;-)

**How do I call the Magento API SOAP V2 with multiple complex filters to the same field?**
In my case I use the [salesOrderList](https://devdocs.magento.com/guides/m1x/api/soap/sales/salesOrder/sales_order.list.html) function.

**This source code does not work:**
```php
$params = array(array(
        'complex_filter' => array(
            array(
                'key' => 'increment_id',
                'value' => array(
                    'key' => 'nlike',
                    'value' => 'a_%'
                ),
            ),
            array(
                'key' => 'increment_id',
                'value' => array(
                    'key' => 'nlike',
                    'value' => 'e_%'
                ),
            ),
        )
    ));

$result = $client->__call('salesOrderList', $params);`
```

**The final sql code only contains:**
`AND (increment_id NOT LIKE 'a_%')`

**But it should contain:**
`AND (increment_id NOT LIKE 'a_%') AND (increment_id NOT LIKE 'e_%')`

For SQL it doesn't matter how many 'LIKE's I apply to a field?!!

**What am I doing wrong?**

**Comment by colinmollenhour** [comment]
Seems like a reasonable expectation. Pull requests are welcome. :)
